### PR TITLE
Use golangci-lint instead of golint

### PR DESCRIPTION
**Pull request by fasmat**
`golint` has been deprecated for a while. In https://github.com/spacemeshos/go-spacemesh we already migrated to `golangci-lint` instead.

This PR updates `post` to also use `golangci-lint` and fixes all complaints that `golangci-lint` has.

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+  print-issued-lines: true
+
+  # print linter name in the end of issue text, default is true
+  print-linter-name: true
+
+# all available settings of specific linters
+linters-settings:
+  staticcheck:
+    # Select the Go version to target. The default is '1.13'.
+    go: "1.18"
```
Comment (pigmej) [inline .golangci.yml]: Shouldn't we run on 1.19?

Comment (countvonzero) [reply .golangci.yml]: go-spacemesh is still using 1.18.
the guiding principal (prior to me joining) is that there should be a good reason (feature we want to use) to upgrade go version. but i don't really feel too strongly.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 name: CI
 
 env:
-  go-version: '1.19'
+  go-version: "1.19"
```
Comment (countvonzero) [inline .github/workflows/ci.yml:4]: but why is this 1.19?

Comment (fasmat) [reply .github/workflows/ci.yml:4]: I tried 1.18. Using it breaks the Windows tests with the following errors:

`ThreadSanitizer failed to allocate 0x000000959000 (9801728) bytes at 0x200d8fbb54000 (error code: 87)`

See e.g. here: https://github.com/spacemeshos/post/actions/runs/3236096321/jobs/5301430586 with 1.19 this does not happen. I don't know if this is a go 1.18 issue or caused by the GCC that is used by Github. See also https://github.com/golang/go/issues/46099

File: .golangci.yml (hunk 1 above)
Comment (fasmat) [reply .golangci.yml]: post breaks on windows when I try to use 1.18. I will update this config to use 1.19 instead.

Comment (fasmat) [reply .golangci.yml]: I created a PR to also update `go-spacemesh` to use Go 1.19: https://github.com/spacemeshos/go-spacemesh/pull/3650

Comment (pigmej) [reply .golangci.yml]: @countvonzero my point was that in a lot of other places here 1.19 was used ;)

**Review by pigmej — APPROVED** [review]
I'm ok with that with the condition that https://github.com/spacemeshos/go-spacemesh/pull/3650 is also ok/merged.

**Pull request merged**